transport/http/inbound_test.go

```diff
@@ -242,6 +242,106 @@ func TestMuxWithInterceptor(t *testing.T) {
 	}
 }
 
+func TestMultipleInterceptors(t *testing.T) {
+	const (
+		yarpcResp  = "YARPC response"
+		healthResp = "health response"
+		userResp   = "user response"
+	)
+	// This should be the underlying yarpc handler.
+	// For the ease of testing, register it last.
+	baseHandler := Interceptor(func(http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			io.WriteString(w, yarpcResp)
+		})
+	})
+	healthInterceptor := Interceptor(func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if r.URL.Path == "/health" {
+				io.WriteString(w, healthResp)
+			} else {
+				h.ServeHTTP(w, r)
+			}
+		})
+	})
+
+	userInterceptor := Interceptor(func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if r.URL.Path == "/user" {
+				io.WriteString(w, userResp)
+			} else {
+				h.ServeHTTP(w, r)
+			}
+		})
+	})
+
+	tests := []struct {
+		msg          string
+		interceptors []InboundOption
+		url          string
+		want         string
+	}{
+		{
+			msg:          "no user interceptor, /yarpc",
+			interceptors: []InboundOption{healthInterceptor},
+			url:          "/yarpc",
+			want:         yarpcResp,
+		},
+		{
+			msg:          "no user interceptor, /health",
+			interceptors: []InboundOption{healthInterceptor},
+			url:          "/health",
+			want:         healthResp,
+		},
+		{
+			msg:          "no user interceptor, /user",
+			interceptors: []InboundOption{healthInterceptor},
+			url:          "/user",
+			want:         yarpcResp,
+		},
+		{
+			msg:          "user interceptor, /yarpc",
+			interceptors: []InboundOption{healthInterceptor, userInterceptor},
+			url:          "/yarpc",
+			want:         yarpcResp,
+		},
+		{
+			msg:          "user interceptor, /health",
+			interceptors: []InboundOption{healthInterceptor, userInterceptor},
+			url:          "/health",
+			want:         healthResp,
+		},
+		{
+			msg:          "user interceptor, /user",
+			interceptors: []InboundOption{healthInterceptor, userInterceptor},
+			url:          "/user",
+			want:         userResp,
+		},
+	}
```

Comment:
Can we add a test case where two interceptors write to the responseWriter? For instance, I1 writes "abcd" and I2 writes "efgh". The output response would be "abcdefgh", with this test we will ensure that the execution is respected in order.